src/Orleans.Core/Core/ClientBuilderExtensions.cs

```diff
@@ -126,6 +128,15 @@ public static IClientBuilder AddClusterConnectionLostHandler(this IClientBuilder
             return builder;
         }
 
+        public static IClientBuilder AddActivityPropagation(this IClientBuilder builder) =>
+            builder.ConfigureServices(services =>
+                {
+                    var listener = new DiagnosticListener("Orleans");
```

Comment:
Maybe it was a bad idea to pass DiagnosticListener into IoC, but aspnetcore [does the same](https://github.com/dotnet/aspnetcore/blob/2c175f7fbddefa4cd6a9f153a4ec8728c8816a05/src/Hosting/Hosting/src/WebHostBuilder.cs#L282).

Reply:
Will they clash, though? On the silo side, they often share containers. In future, the client may also share a container with ASP.NET in some cases (eg, if we add generic host support for the client)

Reply:
We could have a class which inherits from DiagnosticSource and use that, perhaps?

Reply:
It is an interesting question.
Because it is used `TryAddSingletone` it won't override Aspnet core instance and vice versa, so in case of shared container we can have unexpected behavior - name of the diagnostic listener can be "Orleans" or "Microsoft.AspNetCore".

There are two solutions:
1. We can create a fake empty class which is inherit from `DiagnosticListener` like in [other Orleans parts](https://github.com/dotnet/orleans/blob/master/src/Orleans.Core/Networking/NetworkingTrace.cs), 
2. Create a new static listener like in [`DiagnosticsHandler` ](https://github.com/dotnet/runtime/blob/db58dcdb2a3276a61ce7f12b5d20c40426362556/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L277) and use them directly.

Both can solve the problem with registration in DI, but I prefer the second one.  
Which one do you choose?